src/vs/workbench/contrib/languageRuntime/browser/languageRuntimeActions.ts

```diff
@@ -409,57 +409,94 @@ const selectNewLanguageRuntime = async (
 	// Generate quick pick items for runtimes.
 	const runtimeItems: QuickPickItem[] = [];
 
-	if (activeRuntimes.length > 0) {
-		// Add a separator for active sessions.
+	// Add separator for suggested runtimes
+	const suggestedRuntimes = interpreterGroups
+		.map(group => group.primaryRuntime);
+
+	if (suggestedRuntimes.length > 0) {
 		runtimeItems.push({
 			type: 'separator',
-			label: localize('positron.languageRuntime.projectRuntimes', 'Project')
+			label: localize('positron.languageRuntime.suggestedRuntimes', 'Suggested')
 		});
-		// Add active runtimes first and foremost.
-		activeRuntimes.forEach(runtime => {
+
+		suggestedRuntimes.forEach(runtime => {
 			runtimeItems.push({
 				id: runtime.runtimeId,
 				label: runtime.runtimeName,
 				detail: runtime.runtimePath,
 				iconPath: {
 					dark: URI.parse(`data:image/svg+xml;base64, ${runtime.base64EncodedIconSvg}`),
-				},
-				picked: true
+				}
 			});
 		});
 	}
 
 
 	interpreterGroups.forEach(group => {
-		const language = group.primaryRuntime.languageName;
-		// Add separator with language name.
-		runtimeItems.push({ type: 'separator', label: language });
-		// Add primary runtime first.
-		if (group.primaryRuntime.runtimeId !== currentRuntime?.runtimeId && !activeRuntimeIds.has(group.primaryRuntime.runtimeId)) {
-			runtimeItems.push({
-				id: group.primaryRuntime.runtimeId,
-				label: group.primaryRuntime.runtimeName,
-				detail: group.primaryRuntime.runtimePath,
-				iconPath: {
-					dark: URI.parse(`data:image/svg+xml;base64, ${group.primaryRuntime.base64EncodedIconSvg}`),
-				},
-				picked: (group.primaryRuntime.runtimeId === runtimeSessionService.foregroundSession?.runtimeMetadata.runtimeId),
-			});
-		}
-		// Follow with alternate runtimes.
-		group.alternateRuntimes.sort((a, b) => a.runtimeName.localeCompare(b.runtimeName));
+		// Group runtimes by environment type
+		const runtimesByEnvType = new Map<string, ILanguageRuntimeMetadata[]>();
+
 		group.alternateRuntimes.forEach(runtime => {
 			if (runtime.runtimeId !== currentRuntime?.runtimeId && !activeRuntimeIds.has(runtime.runtimeId)) {
-				runtimeItems.push({
-					id: runtime.runtimeId,
-					label: runtime.runtimeName,
-					detail: runtime.runtimePath,
-					iconPath: {
-						dark: URI.parse(`data:image/svg+xml;base64, ${runtime.base64EncodedIconSvg}`),
-					},
-					picked: (runtime.runtimeId === runtimeSessionService.foregroundSession?.runtimeMetadata.runtimeId),
-				});
+				const envType = `${runtime.runtimeSource}`;
+				if (!runtimesByEnvType.has(envType)) {
+					runtimesByEnvType.set(envType, []);
+				}
+				runtimesByEnvType.get(envType)!.push(runtime);
 			}
+
+		});
+
+		const envTypes = Array.from(runtimesByEnvType.keys());
+
+		// Sort runtimes by version (decreasing), then alphabetically
+		envTypes.forEach(envType => {
+			runtimeItems.push({ type: 'separator', label: envType });
+			runtimesByEnvType.get(envType)!
+				.sort((a, b) => {
```

Comment:
This probably shouldn't block this PR, but it would be nice to let each extension handle version-sorting, in case someone does it a different way. Maybe we encode the assumption that `alternateRuntimes` is sorted within the extension before it's sent to Positron? And then this code relies on that sorting.

Reply:
I was able to relax some of the restrictions, so at least the env types are organized how they are passed in rather than solely alphabetically (eg, for Python we have venv at the top, then uv, then pyenv, etc etc). That felt like a nice mix of extension-sorted as well as standardized across languages. We can keep that in mind for followup work though!